src/api/requestsApi.tsx

```diff
@@ -341,13 +342,15 @@ export const useDeleteRequest = () => {
 
 export const useUpdateStage = () => {
   const queryClient = useQueryClient();
+  const addEvent = useAddEvent();
   const { dispatchToast } = useToastController("toaster");
 
   return useMutation(
     ["updateStage"],
     async (request: {
       requestId: number;
       newStage: (typeof STAGES)[number]["key"];
+      eventTitle?: string;
```

Comment:
This should be required, as we are making a mutation, so we should have an eventTitle.  May have to do something in the places where this is set, where you say eventTitle = currentStage.previousEventTitle ?? "Forward Stage Change: Unknown Transition to " + currentStage.next